packages/clay-date-picker/src/ClayDatePicker.tsx

```diff
@@ -0,0 +1,379 @@
+	 */
+	const [currentMonth, setCurrentMonth] = useCurrentMonth(initialMonth);
+
+	/**
+	 * Indicates the time selected by the user.
+	 */
+	const [currentTime, setCurrentTime] = useCurrentTime(timeFormat);
+
+	/**
+	 * The day selected by the user.
+	 */
+	const [daySelected, setDaySelected] = useState<Date>(initialMonth);
+
+	/**
+	 * An array of the weeks and days list for the current month
+	 */
+	const [weeks, setWeeks] = useWeeks(currentMonth, firstDayOfWeek);
+
+	/**
+	 * Flag to indicate if date is expanded.
+	 */
+	const [expanded, setExpanded] = useState(initialExpanded);
+
+	/**
+	 * Create a ref to store the datepicker DOM element
+	 */
+	const elementRef = useRef<HTMLDivElement | null>(null);
+
+	/**
+	 * Handles the click on element of the day
+	 */
+	function handleDayClicked(date: Date) {
+		setDaySelected(date);
+		onValueChange(date);
+	}
+
+	/**
+	 * Handles the input change.
+	 */
+	function inputChange(event: ChangeEvent<HTMLInputElement>) {
+		const {value} = event.target;
+		const format = time ? `${dateFormat} ${timeFormat}` : dateFormat;
+		const date = moment(value, format);
+		const year = date.year();
+
+		if (date.isValid() && year >= years.start && years.end >= year) {
+			setCurrentMonth(date.toDate());
+			setDaySelected(date.toDate());
+
+			if (time) {
+				setCurrentTime(date.hours(), date.minutes());
+			}
+		}
+
+		onValueChange(value);
+	}
+
+	/**
+	 * Handles dot cliecked
+	 */
+	const handleDotClicked = () => {
+		setCurrentMonth(initialMonth);
+		setDaySelected(initialMonth);
+		onValueChange(initialMonth);
+	};
+
+	/**
+	 * Handles datepicker view
+	 */
+	const handleCalendarButtonClicked = () => setExpanded(!expanded);
+
+	/**
+	 * Handles document click
+	 */
+	const handleDocClick = () => setExpanded(false);
+
+	useEffect(() => {
+		// It is not necessary to update the weeks when the native
+		// date is enabled. Unnecessary rendering and processing.
+		if (!useNative) {
+			setWeeks(currentMonth);
+		}
+	}, [currentMonth]);
+
+	return (
+		<div className="date-picker" ref={elementRef}>
+			<div className="input-group" id={id}>
+				<div className="input-group-item">
+					<InputDate
+						ariaLabel={ariaLabels.input}
+						currentTime={currentTime}
+						dateFormat={dateFormat}
+						inputName={inputName}
+						onChange={inputChange}
+						placeholder={placeholder}
+						time={time}
+						timeFormat={timeFormat}
+						useNative={useNative}
+						value={value}
+					/>
+					{!useNative && (
+						<div className="input-group-inset-item input-group-inset-item-after">
+							<Button
+								className="date-picker-dropdown-toggle"
+								icon="calendar"
+								onClick={handleCalendarButtonClicked}
+								spritemap={spritemap}
+								variant="unstyled"
+							/>
+						</div>
+					)}
+				</div>
+			</div>
+
+			{!useNative && (
+				<Dropdown
+					containerRef={elementRef}
+					expanded={expanded}
+					onDocumentClick={handleDocClick}
+				>
+					<React.Fragment>
```

Comment:
In an earlier PR a couple months back I asked whether we could use `<>`/`</>` shorthand for fragments and there was some dependency in the toolchain that wasn't up-to-date enough to cope with it (sadly, I can't remember which one it was); is that still the case now?

Reply:
I think the limitation was babel 6 but I think we can use it now. But I'll remove the Fragment is no longer needed here after we've changed the Date Picker markup. I forgot to remove 😅